pkgs/servers/web-apps/kavita/default.nix

```diff
@@ -25,18 +25,20 @@ stdenvNoCC.mkDerivation (finalAttrs: {
 
     patches = [
       # The webroot is hardcoded as ./wwwroot
-      (substituteAll {
-        src = ./change-webroot.diff;
-        web_root = "${finalAttrs.frontend}/lib/node_modules/kavita-webui/dist";
-      })
+      ./change-webroot.diff
     ];
+    postPatch = ''
+      substituteInPlace API/Services/DirectoryService.cs --replace "@NixKavitaBackendLib@" "$out/lib/kavita-backend/"
+      substituteInPlace API/Startup.cs API/Services/LocalizationService.cs API/Controllers/FallbackController.cs \
+        --replace "@NixKavitaWebRoot@" "${finalAttrs.frontend}/lib/node_modules/kavita-webui/dist/browser"
+    '';
```

Comment:
Please keep the subsituteAll, as it is more clean

Reply:
Do you have an idea for how to refer to the package output rather than the patch output, in that case? I first tried to preserve the `substituteAll` but couldn't figure that part out.

Reply:
I don't think that is possible, as $out or placeholder $out refers to the patch

Reply:
In that case, I have to use the bash `substituteAll` - this seems like a bad choice, as the variables aren't explicitly defined, making it hard to trace what it's used for (further explained in #237216). I cleaned up the `substituteInPlace` usage a little, maybe this is more palatable.